src/Containers/DenseAxisArray.jl

```diff
@@ -60,9 +60,15 @@ end
 
 function Base.getindex(
     x::_AxisLookup{Dict{K,Int}},
-    key::K,
-) where {K<:AbstractVector}
-    return x.data[key]
+    key::T,
+) where {T<:AbstractVector,K>:T}
+    # In this method, we can't tell if `keys` is an actual key, or a vector of
+    # keys. One common cause happens when `K` is `Any`. To ensure correctness,
+    # we need to slightly expensive check if `keys` is a key.
+    if haskey(x.data, key)
+        return x.data[key]
+    end
+    return [x[k] for k in key]
```

Comment:
Can't there be some edge cases here in case the vector of keys is like
`[1, [1], [[1]]]`, Look's like Russel's paradox :laughing:

Reply:
Let's hope that no one builds the indices of their variables using set theory...

I think this is still safe to merge. It's a strict improvement that fixes buggy behavior. No one has every complained about not being able to write a key like `[1, [1], [[1]]]`...

Reply:
`Any[Any["a"], Any["b"], Any[Any["a"], Any["b"]]]`

Reply:
Call is not sure what to do here. Look at `Any[1, 2]`. Disallow slicing if case is ambiguous. Provide alternate slicing syntax.